src/branchandbound.jl

```diff
@@ -1,101 +1,59 @@
 # univariate
-function enclose_BranchandBound(f::Function, dom::Interval; order=10, tol=0.6)
-    x0 = Interval(mid(dom))
-    x = TaylorModel1(order, x0, dom)
-    return branchandbound(f(x-x0).pol, dom, tol)
-end
 
-# multivariate
-function enclose_BranchandBound(f::Function, dom::IntervalBox{D};
-                                order=10, tol=0.6) where {D}
-    x0 = mid(dom)
-    set_variables(Float64, "x", order=2order, numvars=D)
-    x = [TaylorModelN(i, order, IntervalBox(x0), dom) for i=1:D]
-    return branchandbound(f(x...).pol, dom - x0, tol)
+@inline function enclose(f::Function, X::Interval, ba::BranchAndBoundEnclosure;
+                         df=t->ForwardDiff.derivative(f, t))
```

Comment:
```suggestion
                         df=Base.Fix1(ForwardDiff.derivative(f))
```

Reply:
What does this do?

Reply:
it behaves similarly as the df` you had before (but it's a struct); actually i had a typo, just fixed it

Reply:
Cool, can you do sth similar for the gradient?

Reply:
there's only `Base.Fix1` and `Base.Fix2` for partial application with two-argument functions

for the gradient i think it's cleaner if we make it such that `df = Base.Fix1(_gradient, f)` works, for some `_gradient` function that unwraps the domain's components and calls `ForwardDiff.gradient`.

Reply:
That gets pretty noisy already, is it worth it? What is the benefit of `Base.fix1` ?

Reply:
well `df = Base.Fix1(_gradient, f)` is cleaner than `df=t->ForwardDiff.gradient(w->f(w...), t.v))` ? the fact that we need to do two anonymous functions is an implementation detail

Reply:
Once #75 and the IntervalBox thing is fixed I think you could just do
```julia
Base.fix1(ForwardDiff.gradient, f)
```

dont know if it's more readable than
```julia
t->ForwardDiff.gradient(f, t)
```
though

Reply:
yeah for two-arg partial function evaluation i would rather use `Fix1` / `Fix2`

the idea of using `_gradient` was to make things easier in the future -- it's simple to change `_gradient -> gradient` and delete `_gradient`

Reply:
I've committed your suggestion for the derivative, maybe for the gradient we can wait #75 is agreed so that the format for multivariate functions is fixed?